docgen/src/guide/React_native.md

```diff
@@ -6,9 +6,643 @@ category: guide
+* `refine`, the function to call when the end of the page is reach to load more results. 
+
+On the React Native side we are going to take advantage of the [FlatList](https://facebook.github.io/react-native/docs/flatlist.html) to render this infinite scroll. 
+
+Let’s create our own React Native Infinite Hits widget: 
+
+```jsx
+// First, we need to add the connectInfiniteHits connector to our import
+import { connectInfiniteHits } from 'react-instantsearch/connectors';
+// We also need to import the FlatList and other React Native component
+import { StyleSheet, View, FlatList, Image, Text } from 'react-native';
+
+
+// [...]
+
+const Hits = connectInfiniteHits(({ hits, hasMore, refine }) => {
+
+  /* if there are still results, you can 
+  call the refine function to load more */
+  const onEndReached = function() {
+    if (hasMore) {
+      refine();
+    }
+  };
+
+  const keyExtractor = (item, index) => item.objectID;
+
+  return (
+    <FlatList
+      data={hits}
+      onEndReached={onEndReached}
+      keyExtractor={keyExtractor}
+      renderItem={({ item }) => {
+        return (
+          <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+            <Image
+              style={{ height: 100, width: 100 }}
+              source={{ uri: item.image }}
+            />
+            <View style={{ flex: 1 }}>
+              <Text>
+                {item.name}
+              </Text>
+              <Text>
+                {item.type}
+              </Text>
+            </View>
+          </View>
+        );
+      }}
+    />
+  );
+});
+```
+
+We can then call this component in the render of App:
+
+```jsx
+const App = () =>
+  <InstantSearch
+    appId="latency"
+    apiKey="3d9875e51fbd20c7754e65422f7ce5e1"
+    indexName="bestbuy"
+  >
+   <Hits/>
+  </InstantSearch>
+```
+
+You should now be able to see the results and scroll to fetch more of them. Of course depending of the attributes available on your records you might want to change this template. 
+
+In this section we've seen:
+- how to display the results from Algolia
+- how to build an infinite scrolling experience
+
+## Add a SearchBox
+
+Now that we've added the results, we can start querying our index. To do this, we are gonna use the [connectSearchBox](connectors/connectSearchBox.html) connector. 
+
+This connector gives you two interesting properties: 
+
+* `currentRefinement`, the current query
+* `refine`, the function to call when a new query is entered
+
+On the React Native side we are going to take advantage of the [TextInput](http://facebook.github.io/react-native/releases/0.45/docs/textinput.html#textinput) to render this search box. 
+
+```jsx
+// We need to add the connectSearchBox to our import
+import {
+  connectInfiniteHits,
+  connectSearchBox,
+} from 'react-instantsearch/connectors';
+// We need to add the TextInput to our import
+import {
+  StyleSheet,
+  View,
+  FlatList,
+  Image,
+  Text,
+  TextInput,
+} from 'react-native';
+// [...]
```

Comment:
@spinach, I kept the prettier syntax. If there's too many import it just split to one per line. Do you think it's less readable?